Sources/MapboxMaps/Annotations/Generated/CircleAnnotationManager.swift

```diff
@@ -204,15 +213,97 @@ public class CircleAnnotationManager: AnnotationManagerInternal {
+                annotation.isSelected.toggle()
+                tappedAnnotations.append(annotation)
+            }
+            annotations.append(annotation)
+        }
+
+        self.annotations = annotations
+
+        delegate?.annotationManager(
+            self,
+            didDetectTappedAnnotations: tappedAnnotations)
+    }
+
+    private func createDragSourceAndLayer() {
+        var dragSource = GeoJSONSource()
+        dragSource.data = .empty
+        do {
+            try style.addSource(dragSource, id: dragSourceId)
+        } catch {
+            Log.error(forMessage: "Failed to add the source to style. Error: \(error)")
+        }
+
+        do {
+            // copy the existing layer as the drag layer
+            var properties = try style.layerProperties(for: layerId)
+            properties[SymbolLayer.RootCodingKeys.id.rawValue] = dragLayerId
+            properties[SymbolLayer.RootCodingKeys.source.rawValue] = dragSourceId
+
+            try style.addPersistentLayer(with: properties, layerPosition: .above(layerId))
+        } catch {
+            Log.error(forMessage: "Failed to add the layer to style. Error: \(error)")
+        }
+    }
+
+    private func removeDragSourceAndLayer() {
+        do {
+            try style.removeLayer(withId: dragLayerId)
+            try style.removeSource(withId: dragSourceId)
+        } catch {
+            Log.error(forMessage: "Failed to remove drag layer. Error: \(error)")
+        }
+    }
+
+    internal func handleDragBegin(with featureIdentifiers: [String]) {
+        guard let annotation = annotations.first(where: { featureIdentifiers.contains($0.id) }) else { return }
+        createDragSourceAndLayer()
+
+        annotationBeingDragged = annotation
+        annotations.removeAll(where: { $0.id == annotation.id })
+
+        do {
+            try style.updateGeoJSONSource(withId: dragSourceId, geoJSON: .feature(annotation.feature))
+        } catch {
+            Log.error(forMessage: "Failed to update drag source. Error: \(error)")
+        }
+    }
+
+    internal func handleDragChanged(with translation: CGPoint) {
+        guard let annotationBeingDragged = annotationBeingDragged,
+              let offsetPoint = offsetPointCalculator.geometry(for: translation, from: annotationBeingDragged.point) else {
+            return
+        }
+
+        self.annotationBeingDragged?.point = offsetPoint
+        do {
+            try style.updateGeoJSONSource(withId: dragSourceId, geoJSON: .feature(annotationBeingDragged.feature))
+        } catch {
+            Log.error(forMessage: "Failed to update drag source. Error: \(error)")
+        }
+    }
+
+    internal func handleDragEnded() {
+        guard let annotationBeingDragged = annotationBeingDragged else { return }
+        annotations.append(annotationBeingDragged)
+        self.annotationBeingDragged = nil
+
+        // avoid blinking annotation by waiting
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
```

Comment:
Why is it `0.2` and not, for example, `0.35`?

Reply:
if the time is too long you get undesired duplicate annotation effects and if its too short you get blinking. 0.2 is roughly the sweet spot for neither of these effects to occur

Reply:
Should it be different on overloaded systems and/or old devices?

Reply:
I think it may be different for overloaded systems. Happy to ticket this out as a tailwork.